cmd/controller-manager/app/options/options.go

```diff
@@ -107,13 +116,46 @@ func (s *ServerOption) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.InheritOwnerAnnotations, "inherit-owner-annotations", true, "Enable inherit owner annotations for pods when create podgroup; it is enabled by default")
 	fs.Uint32Var(&s.WorkerThreadsForPG, "worker-threads-for-podgroup", defaultPodGroupWorkers, "The number of threads syncing podgroup operations. The larger the number, the faster the podgroup processing, but requires more CPU load.")
 	fs.Uint32Var(&s.WorkerThreadsForGC, "worker-threads-for-gc", defaultGCWorkers, "The number of threads for recycling jobs. The larger the number, the faster the job recycling, but requires more CPU load.")
+	fs.StringSliceVar(&s.Controllers, "controllers", []string{defaultControllers}, fmt.Sprintf("Specify controller gates. Use '*' for all controllers, all knownController: %s ,and we can use "+
+		"'-' to disable controllers, e.g. \"-job-controller,-queue-controller\" to disable job and queue controllers.", knownControllers))
 }
 
-// CheckOptionOrDie check leader election flag when LeaderElection is enabled.
+// CheckOptionOrDie checks the option and returns error if it's invalid
 func (s *ServerOption) CheckOptionOrDie() error {
+	// check controllers option
+	if err := s.checkControllers(); err != nil {
+		return err
+	}
+	// check leader election flag when LeaderElection is enabled.
 	return componentbaseconfigvalidation.ValidateLeaderElectionConfiguration(&s.LeaderElection, field.NewPath("leaderElection")).ToAggregate()
 }
 
+// checkControllers checks the controllers option and returns error if it's invalid
+func (s *ServerOption) checkControllers() error {
```

Comment:
IMO, it's a little complicated and not so readable here, the controller in front has higher priority is just simple and acceptable.
And if this check is truly needed, I think use two maps to check is better.

Reply:
This check only verifies whether the same controller exists at the same time. For example, if "+job-controller" and "-job-controller" exist at the same time. No priority is considered.